spec/predicates/link.md

```diff
@@ -1,76 +1,102 @@
+arbitrary but relevant information in the opaque `environment` and `byproducts`
+fields.
+
+For every step described in the layout of the software supply chain, one (or
+more, depending on the threshold) signed link attestations must be presented.
+
 ## Schema
 
 ```jsonc
 {
   // Standard attestation fields:
-  "_type": "https://in-toto.io/Statement/v0.1",
+  "_type": "https://in-toto.io/Statement/v1",
   "subject": [{ ... }],
 
   // Predicate:
-  "predicateType": "https://in-toto.io/Link/v0.2",
+  "predicateType": "https://in-toto.io/attestation/link/v0.3",
   "predicate": {
     "name": "...",
-    "command": "...",
-    "materials": { ... },
+    "command": [ ... ],
+    "materials": [<ResourceDescriptor>, ...],
     "byproducts": { ... },
     "environment": { ... }
   }
 }
 ```
 
-_(Note: This is a Predicate type that fits within the larger
-[Attestation](../README.md) framework.)_
+### Fields
 
 The `predicate` has the same schema as the link's `signed` field in
-[in-toto 0.9] except:
+[in-toto specification] except:
 
 -   `predicate._type` is omitted.  `predicateType` serves the same purpose.
 -   `predicate.products` is omitted. `subject` serves the same purpose.
+-   `predicate.materials` is updated to a list of `ResourceDescriptors`.
+    Each `ResourceDescriptor` entry MUST include `digest`.
 
-## Converting to old-style links
+### Converting to old-style links
 
-A Link predicate may be converted into an in-toto 0.9 link as follows:
+A Link predicate may be converted into an [in-toto specification] link as
+follows:
 
--   Set `link` to be a copy of `predicate`.
--   Set `link.type` to `"link"`.
+-   Set `link._type` to `"link"`.
+-   Set `link.name`, `link.environment`, and `link.byproducts` using
+    `predicate.name`, `predicate.environment`, and `predicate.byproducts`
+    respectively.
 -   Set `link.products` to be a map from `subject[*].name` to
     `subject[*].digest`.
+-   Set `link.materials` to be a map from `predicate.materials[*].name` to
+    `predicate.materials[*].digest`.
 
 In Python:
 
 ```python
 def convert(statement):
-    assert statement.predicateType == 'https://in-toto.io/Link/v0.2'
-    link = statement.predicate.copy()
+    assert statement.predicateType == 'https://in-toto.io/attestation/link/v0.3'
+    link = {}
     link['_type'] = 'link'
+    link['name'] = statement.predicate['name']
+    link['byproducts'] = statement.predicate['byproducts']
+    link['environment'] = statement.predicate['environment']
+    link['materials'] = {s['name'] : s['digest'] for s in statement.predicate['materials']}
     link['products'] = {s['name'] : s['digest'] for s in statement.subject}
     return link
 ```
 
-## TODO
-
--   [ ] Bump up the in-toto version from 0.9 to 1.0 once
-    [in-toto/docs issue #46](https://github.com/in-toto/docs/issues/46) is
-    resolved.
-
 ## Version History
 
+-   0.3: Updated `materials` to use `ResourceDescriptor`.
```

Comment:
materials seems to have changed from an object to a list of `ResourceDescriptor`, is that right?

also:
* changed TypeURI
* command changed from string to list

Reply:
Good point. I should note that command is currently a string, it was misrepresented in v0.2.

Reply:
Updated the changelog.